cpp/benchmarks/copying/contiguous_split_benchmark.cu

```diff
@@ -40,12 +47,8 @@ void BM_contiguous_split_common(benchmark::State& state,
     splits.push_back(std::min(idx + split_stride, static_cast<cudf::size_type>(num_rows)));
   }
 
-  std::vector<std::unique_ptr<cudf::column>> columns(src_cols.size());
-  std::transform(src_cols.begin(), src_cols.end(), columns.begin(), [](T& in) {
-    auto ret = in.release();
-    ret->null_count();
-    return ret;
-  });
+  for (auto&& c : columns)
+    c->null_count();
```

Comment:
This accepts `columns` by reference, alters it in-place (caching the null counts), and then `std::move`s it when constructing the table -- it seems like an undesired use of move semantics to leave the `columns` argument accepted by reference in a "valid but unspecified state." I would suggest copying this exact snippet from #9860 instead. It has a few improvements that will be needed for `clang-tidy`:

https://github.com/rapidsai/cudf/blob/b4a35c7499403288194a10638c050de9bf869664/cpp/benchmarks/copying/contiguous_split_benchmark.cu#L43-L51